test/metabase/lib/query_test.cljc

```diff
@@ -72,35 +72,103 @@
+                 (lib/current-breakouts)
+                 count)))
+    (testing "removing breakout with dependent should not be allowed"
+      (is (thrown-with-msg?
+            #?(:clj Exception :cljs js/Error)
+            #"Clause cannot be removed as it has dependents"
+            (-> query
+                (lib/append-stage)
+                ;; TODO Should be able to create a ref with lib/field [#29763]
+                (lib/filter (lib/= [:field {:lib/uuid (str (random-uuid)) :base-type :type/Integer} "ID"] 1))
+                (lib/remove-clause 0 (first breakouts)))))
+      (is (thrown-with-msg?
+            #?(:clj Exception :cljs js/Error)
+            #"Clause cannot be removed as it has dependents"
+            (-> query
+                (lib/append-stage)
+                (lib/fields [[:field {:lib/uuid (str (random-uuid)) :base-type :type/Integer} "ID"]])
+                (lib/append-stage)
+                ;; TODO Should be able to create a ref with lib/field [#29763]
+                (lib/filter (lib/= [:field {:lib/uuid (str (random-uuid)) :base-type :type/Integer} "ID"] 1))
+                (lib/remove-clause 0 (first breakouts)))))
       (is (= 0 (-> query
-                   (lib/remove-clause (first order-bys))
-                   (lib/remove-clause (second order-bys))
-                   (lib/order-bys)
+                   (lib/remove-clause 0 (second breakouts))
+                   (lib/append-stage)
+                   ;; TODO Should be able to create a ref with lib/field [#29763]
+                   (lib/filter (lib/= [:field {:lib/uuid (str (random-uuid)) :base-type :type/Integer} "ID"] 1))
+                   (lib/remove-clause 0 (first breakouts))
+                   (lib/current-breakouts 0)
                    count))))))
 
-(deftest ^:parallel replace-clause-test
+(deftest ^:parallel replace-clause-order-by-test
   (let [query (-> (lib/query-for-table-name meta/metadata-provider "VENUES")
                   (lib/filter (lib/= "myvenue" (lib/field (meta/id :venues :name))))
                   (lib/order-by (lib/field (meta/id :venues :name)))
                   (lib/order-by (lib/field (meta/id :venues :name))))
         order-bys (lib/order-bys query)]
-    (testing "order-bys"
-      (is (= 2 (count order-bys)))
-      (let [replaced (-> query
-                         (lib/replace-clause (first order-bys) (lib/order-by-clause (lib/field (meta/id :venues :id)))))
-            replaced-order-bys (lib/order-bys replaced)]
-        (is (not= order-bys replaced-order-bys))
-        (is (= 2 (count replaced-order-bys)))
-        (is (= (second order-bys) (second replaced-order-bys)))))))
+    (is (= 2 (count order-bys)))
+    (let [replaced (-> query
+                       (lib/replace-clause (first order-bys) (lib/order-by-clause (lib/field (meta/id :venues :id)))))
+          replaced-order-bys (lib/order-bys replaced)]
+      (is (not= order-bys replaced-order-bys))
```

Comment:
Can't we have a more precise assertion about the first (replaced) order-by?
Would replacing the second order-by work as expected?